run-tests.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh --
+set -eu
+case $0 in
+(/*) cd "${0%/*}/";;
+(*/*) cd "./${0%/*}";;
+esac
```

Comment:
Why not simply `cd "$(dirname "$0")"`? much shorter _and_ more readable.

Reply:
This breaks if dirname fails (for some odd reason) or if the string it returns ends with a newline (ugh).  I believe the code I am using here is unconditionally correct; I consider it to be practically an idiom in shell scripting.

Reply:
> This breaks if dirname fails (for some odd reason)

`set -e` a line before -> not an issue

> the string it returns ends with a newline (ugh)

Last time I checked, `app-linux-split-gpg2` does not end with a newline. 

Anyway, please don't. Perhaps your version might better handle never-to-happen cases, but at the cost of readability for all the 99.99999% other cases.